storage/database/db_manager.go

```diff
@@ -1797,6 +1816,44 @@ func (dbm *databaseManager) WriteChainConfig(hash common.Hash, cfg *params.Chain
 	}
 }
 
+// ReadAccountSnapshot retrieves the snapshot entry of an account trie leaf.
+func (dbm *databaseManager) ReadAccountSnapshot(hash common.Hash) []byte {
+	db := dbm.getDatabase(SnapshotDB)
+	data, _ := db.Get(AccountSnapshotKey(hash))
+	return data
+}
+
+// WriteAccountSnapshot stores the snapshot entry of an account trie leaf.
+func (dbm *databaseManager) WriteAccountSnapshot(hash common.Hash, entry []byte) {
+	db := dbm.getDatabase(SnapshotDB)
+	writeAccountSnapshot(db, hash, entry)
+}
+
+// DeleteAccountSnapshot removes the snapshot entry of an account trie leaf.
+func (dbm *databaseManager) DeleteAccountSnapshot(hash common.Hash) {
+	db := dbm.getDatabase(SnapshotDB)
+	deleteAccountSnapshot(db, hash)
+}
+
+// ReadStorageSnapshot retrieves the snapshot entry of an storage trie leaf.
+func (dbm *databaseManager) ReadStorageSnapshot(accountHash, storageHash common.Hash) []byte {
+	db := dbm.getDatabase(SnapshotDB)
+	data, _ := db.Get(StorageSnapshotKey(accountHash, storageHash))
```

Comment:
Why don't we implement readStorageSnapshot()?

Reply:
The `writeStorageSnapshot` is for batch write as well as a single(?) write. Do you think `readStorageSnapshot` is needed?